website/docs/migration/from_state_notifier.mdx

```diff
@@ -0,0 +1,264 @@
+a `CancelToken` can be used to cancel requests.
+
+Therefore, the above example migrates to the following:
+<AutoSnippet language="dart" {...oldLifecyclesFinal}></AutoSnippet>
+
+
+## Mutations APIs are the same as before
+
+Up until now we've shown the differences between `StateNotifier` and the new APIs.  
+Instad, one thing `Notifier`, `AsyncNotifer` and `StateNotifier` share is how their states
+can be consumed and mutated.
+
+Consumers can obtain data from these three providers with the same syntax, which is great in case
+you're migrating away from `StateNotifier`, and this applies for notifiers methods, too.
+
+<AutoSnippet language="dart" {...consumersDontChange}></AutoSnippet>
+
+## Other migrations
+
+Let's explore the less-impactful differences between `StateNotifier` and `Notifier` (or `AsyncNotifier`)
+
+### From `.addListener` and `.stream`
+
+`StateNotifier`'s `.addListener` and `.stream` can be used to listen for state changes.
+These two APIs are now to be considered outdated.
+
+This is intentional due to the desire to reach full API uniformity with `Notifier`, `AsyncNotifier` and other providers.  
+Indeed, using a `Notifier` or an `AsyncNotifier` shouldn't be any different from any other provider.
+
+Therefore this:
+<AutoSnippet language="dart" {...addListenerOld}></AutoSnippet>
+
+Becomes this:
+<AutoSnippet language="dart" {...addListener}></AutoSnippet>
+
+In a nutshell: if you want to listen to a `Notifier`/`AsyncNotifer`, just use `ref.listen`,
+[as explained here](/docs/concepts/reading#using-reflisten-to-react-to-a-provider-change).
+
+### From `.debugState` in tests
+
+`StateNotifier` exposes `.debugState`: this property is used for pkg:state_notifier users to enable
+state access from outside the class when in development mode, for testing purposes.
+
+If you're using `.debugState` to access state in tests, chances are that you need to drop this
+approach.
+
+`Notifier` / `AsyncNotifer` don't have a `.debugState`; instead, they expose `.state`, which is
+`@visibleForTesting`.
+
+Nonetheless, accessing this property from tests is viable _if and only if_ you had a `Notifier` /
+`AsyncNotifer` properly instantied; then, you could access `.state` inside tests freely.
+
+Indeed, `Notifier` / `AsyncNotifier` should _not_ be instantiated by hand: they should instead be
+interacted with by using its provider.  
+Failing to do so will *break* the notifier, due to ref and family args not being initialized.
+
+Don't have a `Notifier` instance? No problem, you can read its provider to obtain one.  
+
+If you want to interact with `Notifier`/`AsyncNotifier` in a testing environment, there's a good
+chance you should do so just like with any other provider: by reading them directly (as opposed of
+directly instantiating a `Notifier` by hand).
+
+The [testing guide](/docs/cookbooks/testing) fully explains how to use `ProviderContainer.read` to do so.
```

Comment:
that link is out of date.
Use the new testing page instead  https://riverpod.dev/docs/essentials/testing